File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"
#include <stdio.h>
#include <math.h>

unsigned int sum_of_squares(unsigned int number) {
    if (number == 1) return 1;
    unsigned int gauss_sqrt = (number*(number+1)*(2*number+1))/6;
    return gauss_sqrt;
}

unsigned int square_of_sum(unsigned int number){
    if (number == 1) return 1;
    unsigned int gauss = (number*(number+1))/2;
    return pow(gauss, 2);
}

unsigned int difference_of_squares(unsigned int number){
    return square_of_sum(number) - sum_of_squares(number);
}

/*int main() {
    printf("%u", difference_of_squares(6));
    return 0;
}*/

File: zig/difference-of-squares/difference_of_squares.zig
const std = @import("std");

pub fn squareOfSum(number: usize) usize {
    const calc_num: usize = @divTrunc(number * (number + 1), 2);
    return calc_num * calc_num;
}

pub fn sumOfSquares(number: usize) usize {
    const calc_num: usize = number * (number + 1) * (2 * number + 1);
    return @divTrunc(calc_num, 6);
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
